quickannotator/api/v1/tile/__init__.py

```diff
@@ -25,134 +24,125 @@ class PredictTileRespSchema(Schema):
     @bp.arguments(GetTileArgsSchema, location='query')
     @bp.response(200, TileBoundingBoxRespSchema)
-    def get(self, args):
+    def get(self, args, image_id, annotation_class_id):
         """     get the bounding box for a given tile
         """
-        tilespace = get_tilespace(image_id=args['image_id'], annotation_class_id=args['annotation_class_id'], in_work_mag=False)
+        tilespace = get_tilespace(image_id=image_id, annotation_class_id=annotation_class_id, in_work_mag=False)
         bbox = tilespace.get_bbox_for_tile(args['tile_id'])
         return {'bbox': bbox}, 200
 
-@bp.route('/search/bbox')
-class TileSearch(MethodView):
+@bp.route('/<int:image_id>/<int:annotation_class_id>/search/bbox')
+class TileIdSearchByBbox(MethodView):
     @bp.arguments(SearchTileArgsSchema, location='query')
-    @bp.response(200, TileRespSchema(many=True))
-    def get(self, args):
+    @bp.response(200, TileIdRespSchema)
+    def get(self, args, image_id, annotation_class_id):
         """     get all Tiles within a bounding box
         """
-        
-        tilespace = get_tilespace(image_id=args['image_id'], annotation_class_id=args['annotation_class_id'], in_work_mag=False)
+        tilespace = get_tilespace(image_id=image_id, annotation_class_id=annotation_class_id, in_work_mag=False)
         tile_ids_in_bbox = tilespace.get_tile_ids_within_bbox((args['x1'], args['y1'], args['x2'], args['y2']))
-        tile_ids_in_mask, _, _ = get_tile_ids_intersecting_mask(args['image_id'], args['annotation_class_id'], mask_dilation=1)
+        tile_ids_in_mask, _, _ = get_tile_ids_intersecting_mask(image_id, annotation_class_id, mask_dilation=1)
         ids = set(tile_ids_in_bbox) & set(tile_ids_in_mask)
-        query = db_session.query(models.Tile).filter(
-            models.Tile.tile_id.in_(ids),
-            models.Tile.image_id == args['image_id'],
-            models.Tile.annotation_class_id == args['annotation_class_id']
-        )
-
-        if 'hasgt' in args:
-            query = query.filter(models.Tile.hasgt == args['hasgt'])
-        
-        tiles = query.all()
-
-        if args['include_placeholder_tiles']:
-            existing_ids = set([tile.tile_id for tile in tiles])
-            placeholder_tile_ids = ids - existing_ids
-            placeholder_tiles = [models.Tile(tile_id=tile_id, image_id=args['image_id'], annotation_class_id=args['annotation_class_id'], seen=TileStatus.UNSEEN, hasgt=False) for tile_id in placeholder_tile_ids]
-            
-            tiles.extend(placeholder_tiles)
-        return tiles, 200
+
+        # Filter by tiles which have ground truths saved. This limits the number of tiles we have to consider for rendering.
+        if args['hasgt']:
+            ids = db_session.query(models.Tile.tile_id).filter(
+                models.Tile.annotation_class_id == annotation_class_id,
+                models.Tile.image_id == image_id,
+                models.Tile.tile_id.in_(ids),
```

Comment:
tile_id.in_(ids) <--- is this correct? is the second id the row id or the tile_id?